File: tests/rustdoc-js/generics-nested.js
// exact-check

const EXPECTED = [
    {
        'query': '-> Out<First<Second>>',
        'others': [
            { 'path': 'generics_nested', 'name': 'alef' },
        ],
    },
    {
        'query': '-> Out<Second<First>>',
        'others': [],
    },
    {
        'query': '-> Out<First, Second>',
        'others': [
            { 'path': 'generics_nested', 'name': 'bet' },
        ],
    },
    {
        // can't put generics out of order
        'query': '-> Out<Second, First>',
        'others': [],
    },
];

File: tests/rustdoc-js/generics-nested.rs
pub struct Out<A, B = ()> {
    a: A,
    b: B,
}

pub struct First<In = ()> {
    in_: In,
}

pub struct Second;

// Out<First<Second>>
pub fn alef() -> Out<First<Second>> {
    loop {}
}

pub fn bet() -> Out<First, Second> {
    loop {}
}
